Design: 매거진 첫 페이지 프로필 UI 스타일링

JJongsKim opened the pull request:
## 📮 관련 이슈
- Resolved: #이슈번호

## ⛳️ 작업 내용

각 매거진 진입 시 보여줄 프로필 UI를 완성했습니다.

## 🌱 PR 포인트

1/6(금) 스프린트 회의 때, 페이지 descripon 부분은 프로필에 적용된 소개로 우선 가져온 뒤 매거진 별로 소개글을 수정할 수 있도록 하자는 의견이 나왔습니다.
🛠 해당 부분은 추후 코드 수정이 필요합니다.

## 📸 스크린샷
|스크린샷|
|:--:|
|![ezgif com-gif-maker](https://user-images.githubusercontent.com/81777778/211156974-6db356ec-f3e5-4b8b-b6f3-6e6b9a28e911.gif)|

## 📎 레퍼런스

<!-- 참고한 레퍼런스가 있다면 기록해주세요 -->

JJongsKim committed:
:lipstick: Design: 매거진 첫 페이지 프로필 UI 틀 스타일링

JJongsKim committed:
:sparkles: Feat: 매거진 첫 페이지 프로필 UI로 변경

- TODO: description 부분은 추후 매거진 별로 수정가능하도록 코드 수정하기

JJongsKim committed:
:bug: Fix: 매거진 게시글이 없을 경우의 UI 수정

JJongsKim committed:
Merge branch 'dev' into design/magazine-1st-profile

elbica commented on src/containers/magazine/PageViewContainer.tsx:
```diff
@@ -21,32 +22,43 @@ const PageViewContainer = ({ pages = PAGES }: Props) => {
-            </li>
-          ))}
-        </ol>
+        {pages.length === 0 ? (
+          <MagazinePageProfile {...PROFILE} />
+        ) : (
```
return에서 삼항연산자로 처리하는 것보다 early return 하는게 갠적으로 코드 읽기가 편해보입니다!
`if(pages.length === 0) return <MagazinePageProfile {...PROFILE} />`

JJongsKim replied:
오호 그 부분을 그렇게 표현할 수 있었겠네요!
남겨주신 코멘트 참고하여 `<ol></ol>` 내부로 이동시켰습니다!

elbica commented on src/containers/magazine/PageViewContainer.tsx:
```diff
@@ -21,32 +22,43 @@ const PageViewContainer = ({ pages = PAGES }: Props) => {
+          <ol css={CSSCarouselContainer}>
+            {pages.map((page, idx) => (
+              <li key={page.page_id} id={`${idx}`} css={CSSCarouselItem}>
+                {idx === 0 ? (
+                  <MagazinePageProfile {...PROFILE} />
+                ) : (
```
idx가 0일 때 profile만 보여준다면 원래 `pages` 배열에 있던 `pages[0]` 데이터는 안보일것 같은데 체크해 보셨을까요?

```tsx
<ol css={CSSCarouselContainer}>
 <li css={...}> <MagazinePageProfile {...PROFILE} /> </li>
 { pages.map(...) }
</ol>
```
- 이런식으로 조건문을 줄이는 것은 어떨까요?

elbica reviewed (commented):
고생하셨습니다😊
코멘트 확인 부탁드립니다~!

JJongsKim committed:
:bug: Fix: hash Link 데이터 모두 볼 수 있도록 수정

- 프로필 UI가 맨 앞으로 올 수 있도록 코드 수정
- 대신 매거진 페이지 끝에서 다시 프로필로 돌아오거나, 첫 번째 페이지에서 이전 버튼을 눌렀을 때 프로필로 돌아오지 않음(수정 필요)
